game/game.go

```diff
@@ -562,6 +562,62 @@ func (g *Game) performMove(m *move) (bool, string) {
 	return true, tgtPiece
 }
 
+// Some logic is duplicated with the function performMove. Look into reusing logic where possible.
+func forecastMove(currentBoard [][]string, pieceToInt map[string]int, canAttack [][]bool, srcFile, srcRank, tgtFile, tgtRank int) (bool, [][]string) {
```

Comment:
If this function is going to be exported and accessible outside the package (eg, for anAI,) it needs to be capitalized. `ForecastMove`.